scripts/setup.sh

```diff
@@ -0,0 +1,290 @@
+    esac
+    local os="linux"
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    
+    echo "Downloading Go ${version}..."
+    if ! curl -L "https://go.dev/dl/go${version}.${os}-${arch}.tar.gz" -o "${temp_dir}/go.tar.gz"; then
+        echo "Error: Failed to download Go" >&2
+        rm -rf "$temp_dir"
+        exit 1
+    fi
+
+    echo "Verifying checksum..."
+    local checksum_url="https://go.dev/dl/go${version}.${os}-${arch}.tar.gz.sha256"
+    local expected_sum
+    expected_sum=$(curl -sL "$checksum_url" | awk '{print $1}')
+    local actual_sum
+    actual_sum=$(sha256sum "${temp_dir}/go.tar.gz" | awk '{print $1}')
+    
+    if [[ $expected_sum != $actual_sum ]]; then
+        echo "Error: Checksum mismatch for Go archive" >&2
+        rm -rf "$temp_dir"
+        exit 1
+    fi
+    
+    echo "Installing Go ${version}..."
+    if ! rm -rf /usr/local/go && tar -C /usr/local -xzf "${temp_dir}/go.tar.gz"; then
+        echo "Error: Failed to install Go" >&2
+        rm -rf "$temp_dir"
+        exit 1
+    fi
+    
+    rm -rf "$temp_dir"
+    
+    if ! grep -q "/usr/local/go/bin" /etc/profile.d/go.sh 2>/dev/null; then
+        echo 'export PATH=$PATH:/usr/local/go/bin' >> /etc/profile.d/go.sh
+        chmod +x /etc/profile.d/go.sh
+        source /etc/profile.d/go.sh
+    fi
+}
+
+# check if the go version is installed else install it 1.23.4
+function check_go_version() {
+    if ! command -v go &>/dev/null; then
+        echo "Go is not installed. Installing..."
+        install_go
+    fi
+}
+
+# install air hot reload for golang
+function install_air_hot_reload(){
+    local user_home
+    user_home=$(eval echo ~${SUDO_USER:-$USER})
+    GOPATH="$user_home/go" go install github.com/air-verse/air@latest
+    export PATH="$PATH:$user_home/go/bin"
+}
+
+# load the env variables from the api/.env.sample file
+function load_api_env_variables(){
+    move_to_folder "api"
+    if [ -f .env.sample ]; then
+        while IFS='=' read -r key value; do
+            # Skip empty lines and comments
+            [[ -z "$key" || "$key" =~ ^# ]] && continue
+            # Remove any quotes from the value
+            value=$(echo "$value" | tr -d '"'"'")
+            # Export the variable
+            export "$key=$value"
+        done < .env.sample
+    else
+        echo "Error: .env.sample file not found in api directory" >&2
+        exit 1
+    fi
+    move_to_folder ".."
+}
+
+# setup postgres with docker
+function setup_postgres_with_docker(){
+    load_api_env_variables
+    docker run -d --name nixopus-db \
+        -e POSTGRES_USER="${USERNAME:-nixopus}" \
+        -e POSTGRES_PASSWORD="${PASSWORD:-nixopus}" \
+        -e POSTGRES_DB="${DB_NAME:-nixopus}" \
+        -p "${DB_PORT:-5432}:5432" \
+        postgres
+    echo "Postgres setup completed successfully"
+}
+
+# setup ssh will create a ssh key and add it to the authorized_keys file
+function setup_ssh(){
+    # TODO: generate SSH key and add to authorized_keys
+    return 0
```

Comment:
For now should we add logs to let user know that they should manually update ssh key as part of post installation logs?